src/anomalib/metrics/threshold/mebin.py

```diff
@@ -0,0 +1,250 @@
+
+Example:
+    >>> from anomalib.metrics.threshold import MEBin
+    >>> import numpy as np
+    >>> # Create sample anomaly maps
+    >>> anomaly_maps = [np.random.rand(256, 256) * 255 for _ in range(10)]
+    >>> # Initialize and compute thresholds
+    >>> mebin = MEBin(anomaly_map_list=anomaly_maps, sample_rate=4)
+    >>> binarized_maps, thresholds = mebin.binarize_anomaly_maps()
+    >>> print(f"Computed {len(thresholds)} thresholds")
+
+Note:
+    The algorithm works best when anomaly maps contain clear separation between
+    normal and anomalous regions. The min_interval_len parameter should be tuned
+    based on the expected stability of anomaly score distributions.
+"""
+
+import cv2
+import numpy as np
+from tqdm import tqdm
+
+
+
+class MEBin:
+    """MEBin adaptive thresholding algorithm for anomaly detection.
+
+    This class implements the MEBin (Minimum Entropy Binarization) algorithm
+    which automatically determines optimal thresholds for converting continuous
+    anomaly maps to binary masks by analyzing the stability of connected
+    component counts across different threshold levels.
+
+    The algorithm works by:
+        - Sampling anomaly maps at configurable rates across threshold range
+        - Counting connected components at each threshold level
+        - Identifying stable intervals where component count remains constant
+        - Selecting the optimal threshold from the longest stable interval
+        - Optionally applying morphological erosion to reduce noise
+
+    Args:
+        anomaly_map_path_list (list, optional): List of file paths to anomaly maps.
+            If provided, maps will be loaded as grayscale images. 
+            Defaults to None.
+        anomaly_map_list (list, optional): List of anomaly map arrays. If provided,
+            maps should be numpy arrays. 
+            Defaults to None.
+        sample_rate (int, optional): Sampling rate for threshold search. Higher
+            values reduce processing time but may affect accuracy. 
+            Defaults to 4.
+        min_interval_len (int, optional): Minimum length of stable intervals.
+            Should be tuned based on the expected stability of anomaly score
+            distributions. 
+            Defaults to 4.
+        erode (bool, optional): Whether to apply morphological erosion to
+            binarized results to reduce noise. 
+            Defaults to True.
+
+    Example:
+        >>> from anomalib.metrics.threshold import MEBin
+        >>> import numpy as np
+        >>> # Create sample anomaly maps
+        >>> anomaly_maps = [np.random.rand(256, 256) * 255 for _ in range(10)]
+        >>> # Initialize MEBin
+        >>> mebin = MEBin(anomaly_map_list=anomaly_maps, sample_rate=4)
+        >>> # Compute binary masks and thresholds
+        >>> binarized_maps, thresholds = mebin.binarize_anomaly_maps()
+    """
+
+    def __init__(self, anomaly_map_path_list=None, sample_rate=4, min_interval_len=4, erode=True):
```

Comment:
What if we want to use this post-processing in an end-to-end operation in anomalib? Would we need to follow this logic? 
1. Save anomaly maps to file
2. Load them here and apply post-processing? 

Would it not be possible to use this approach without loading the heatmaps from file?